src/Microsoft.Extensions.Logging.Abstractions/Internal/LogValuesFormatter.cs

```diff
@@ -18,24 +18,46 @@ public class LogValuesFormatter
         private const string NullValue = "(null)";
         private static readonly object[] EmptyArray = new object[0];
         private static readonly char[] FormatDelimiters = {',', ':'};
-        private readonly string _format;
-        private readonly List<string> _valueNames = new List<string>();
+        private string _format;
+        private List<string> _valueNames = new List<string>();
```

Comment:
Do we need to eagerly allocate _valueNames?

Reply:
The allocation is modified to be initialized when there are braces in the input format and suitable null checks are added.